### DISCUSSION TITLE
create user in another segment programmatically

### DISCUSSION THREAD

**Discussion by nyemiks** [opencrx/opencrx-documentation#36] [Q&A] [answered]
I have an opencrx user with log in credentials in one segment. How can I create a user with the same log in credentials on another segment. I want to use the same login id on both segments.

is there a back end class I can re use for this ?

**Comment by wfro64** [comment, marked as answer]
There are two API functions to create a user:

- home1:Segment::createUser
- home1::Segment::importUsers

For more information see [here](https://www.opencrx.org/opencrx/5.2/uml/opencrx-core/home1/tn/999_Segment.png.html). 

These functions create a segment-specific user home and a user principal. The login principal and credentials are shared and stored in the "Root" segment.

The corresponding backend class is org.opencrx.kernel.backend.UserHomes.

**Reply by nyemiks** [reply]
I have looked at the import users method. It expects a binary. How should the data be structured before conversion to binary ? is it expecting comma separated values (csv) 

What fields should be specified ?
